### DISCUSSION TITLE
Should ListFeed operators been suffixed by "Item"

### DISCUSSION THREAD

**Discussion by dr1rrb** [unoplatform/uno.extensions#1316] [General]
Currently on `ListFeed` the single operator is `.Where` and is about item itself (i.e. `ListFeed<T>.Where(Predicate<T>)`). As we are about to add `.Select` and `.SelectAsync` the question is should we suffix operators by `Item` to make distinction clearer (`.WhereItem`, .`SelectItem`, `.SelectItemAsync`).

Reminder: The `IListFeed<T>` **does not** implements `IFeed<T>` so there is no possible conflict between `ListFeed<T>.Where(Predicate<T>)` and `ListFeed<T>.Where(Predicate<IImmutableList<T>>)` (which does not exists and is not planned).

Pros:
* Make distinction clearer between `Feed` and `ListFeed`
* Even if not planned at all, it offers the ability to add collection level operator (e.g. `ListFeed<T>.Where(collection => collection.Count > 0)`).

Cons:
* Breaks usage of linq-syntax
* `.Where` and `.Select` are from LINQ which is already about list and collections, i.e. the "item" implicit.

**Comment by nickrandolph** [comment]
I would prefer to stick with Select, Where (ie without the Item suffix) as this is more familiar with what developers are used to coming from more traditional list based linq operations

**Comment by weitzhandler** [comment]
My ¢2: I'd second @nickrandolph and stick with `Select` etc. (no suffix). The ones with `Item` suffix is adding another layer of confusion and unknown.
